### PR TITLE
feat: [CDS-77772]: supprt for params and new env variables

### PR DESCRIPTION
**Pull request by tmacari-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51981)

### CODE REVIEW

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger messagecheck

**Comment by tmacari-harness** [comment]
static schema PR: https://github.com/harness/harness-schema/pull/121

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger smartchecks

File: 125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/AwsCdkDeployPluginInfoProvider.java [hunk 1]
```diff
@@ -67,7 +71,18 @@ private Map<String, String> getEnvironmentVariables(AwsCdkDeployStepInfo awsCdkD
     if (isNotEmpty(stackNames)) {
       environmentVariablesMap.put(PLUGIN_AWS_CDK_STACK_NAMES, String.join(" ", stackNames));
     }
-
+    Map<String, String> parameters = getParameterFieldValue(awsCdkDeployStepInfo.getParameters());
+    List<String> parametersList = new ArrayList<>();
+    if (isNotEmpty(parameters)) {
```
Comment (vlica-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/AwsCdkDeployPluginInfoProvider.java:76]: @tmacari-harness
Would it be possible to have this if condition into a method?
I see it's used the same in AwsCdkDeployPluginInfoProvider.java and AwsCdkDeployStep.java

**Comment by tmacari-harness** [comment]
trigger uts

**Review by ivan-83 — APPROVED** [review]
LGTM with minor comment related to renaming. 

> Reviewed 4 of 12 files at r1, 2 of 3 files at r2, 5 of 7 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51981#-:-Ncd5uDY0JYvZCGL17GD:b-nqvdoi)* status: 14 file(s) to review. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/provision/awscdk/AwsCdkDeployStepInfo.java` line 102 at r4](https://reviewable.io/reviews/harness/harness-core/51981#-Ncd58GkE-emRdlVXcPT:-Ncd58GkE-emRdlVXcPU:b48kti8) ([raw file](https://github.com/harness/harness-core/blob/68a50c6ae7f8dfbf6d5e156b42360512ca19debf/125-cd-nextgen/src/main/java/io/harness/cdng/provision/awscdk/AwsCdkDeployStepInfo.java#L102)):*
> ```Java
>         .envVariables(getEnvVariables())
>         .stackNames(getStackNames())
>         .parameters(getParameters())
> ```

Minor - What are parameters related to, just asking because of the property name. The image, stackName, and provisionerIdentifier are also step parameters.

**Comment by tmacari-harness** [comment]
trigger ut4

**Comment by tmacari-harness** [comment]
trigger ut7

**Comment by tmacari-harness** [comment]
trigger ut9

**Comment by tmacari-harness** [comment]
AWS CDk or cloudformation parameteres https://docs.aws.amazon.com/cdk/v2/guide/parameters.html

**Comment by tmacari-harness** [comment]
> LGTM with minor comment related to renaming.
> 
> > Reviewed 4 of 12 files at r1, 2 of 3 files at r2, 5 of 7 files at r3, 3 of 3 files at r4, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/51981#-:-Ncd5uDY0JYvZCGL17GD:b-nqvdoi)_ status: 14 file(s) to review.
> 
> _[`125-cd-nextgen/src/main/java/io/harness/cdng/provision/awscdk/AwsCdkDeployStepInfo.java` line 102 at r4](https://reviewable.io/reviews/harness/harness-core/51981#-Ncd58GkE-emRdlVXcPT:-Ncd58GkE-emRdlVXcPU:b48kti8) ([raw file](https://github.com/harness/harness-core/blob/68a50c6ae7f8dfbf6d5e156b42360512ca19debf/125-cd-nextgen/src/main/java/io/harness/cdng/provision/awscdk/AwsCdkDeployStepInfo.java#L102)):_
> 
> > ```java
> >         .envVariables(getEnvVariables())
> >         .stackNames(getStackNames())
> >         .parameters(getParameters())
> > ```
> 
> Minor - What are parameters related to, just asking because of the property name. The image, stackName, and provisionerIdentifier are also step parameters.

**Comment by tmacari-harness** [comment]
@ivan-83 :  
AWS CDk or cloudformation parameteres https://docs.aws.amazon.com/cdk/v2/guide/parameters.html

**Comment by tmacari-harness** [comment]
trigger review

**Comment by tmacari-harness** [comment]
trigger ut7

**Comment by tmacari-harness** [comment]
trigger ut4

**Comment by tmacari-harness** [comment]
trigger ut7

**Pull request merged**